### ISSUE TITLE
Not real time safe mysofa_getfilter_float function

### ISSUE THREAD

**Issue by Bk8** [hoene/libmysofa#50]
I was looking an this is great library, but making an c non real time safe implementation to get filters is a little bit pointless. Filters functions cannot be used in an audio thread because of the great use of malloc, also predefining the filter length is useless if another auxiliar allocation is going to be made inside the function.

//Outside function 
float leftIR[filter_length];
float rightIR[filter_length];

mysofa_getfilter_float(hrtf, 10, 10, 10, leftIR, rightIR, &leftDelay, &rightDelay);

//Inside function 
MYSOFA_EXPORT void mysofa_getfilter_float(struct MYSOFA_EASY* easy, float x, float y, float z,
					  float *IRleft, float *IRright,
					  float *delayLeft, float *delayRight)
{
	float c[3];
	float *fir = malloc(easy->hrtf->N * easy->hrtf->R * sizeof(float)); <--- this can be preacllocated when the sofa file is loaded

**Comment by hoene** [comment]
good point.

Do you want to provide a patch?

**Comment by Bk8** [comment]
Sorry for the late reply, I will submit a patch but I'm trying to understand how it works internally. I'm not really a functional programmer I prefer C++. But I think the best solution is to add this:'

float *fir = malloc(easy->hrtf->N * easy->hrtf->R * sizeof(float));

to MYSOFA_EASY struct or send it as a pointer to the get filter function. Also I don't really know how to remove mallocs from the K-D tree, I'm not sure if allocation is needed every time that a node is found. Finally I found some bugs working under OsX so I'm going to open another issue with that.

**Comment by hoene** [comment]
Please check cfae6a3d1198b8dcef1661b6b0d31faed3f9a70b

**Comment by Bk8** [comment]
Yes, that is a good solution. Thanks! I'm going to try to make a fix for kd-tree since it has lots of mallocs when mysofa_getfilter_float is called.  If you can add some comments in the code it will super useful.

**Comment by hoene** [comment]
Please check https://github.com/hoene/libmysofa/commit/935c5b3c9c7a12e4215a22375eeb16996ae01504
and whether is use more real-time friendly...
